analysis/ahmed/phase_change_filter.py

```diff
@@ -0,0 +1,172 @@
+'''
+phase_change_filter.py - filter BAM for reads containing phase changes
+
+usage:
+    python3.5 phase_change_filter.py \
+            --filename input.bam \
+            --vcf parental.vcf.gz \
+            --out output.bam
+'''
+
+import argparse
+import pysam
+from cyvcf2 import VCF
+from tqdm import tqdm
+from Bio import SeqIO
+
+def args():
+    parser = argparse.ArgumentParser(
+        description='', 
+        usage='python3.5 phase_change_filter.py [options]')
+
+    parser.add_argument('-f', '--filename', required=True,
+                        type=str, help='BAM to filter')
+    parser.add_argument('-v', '--vcf', required=True,
+                        type=str, help='VCF containing parents')
+    parser.add_argument('-o', '--out', required=True,
+                        type=str, help='File to write to')
+
+    args = parser.parse_args()
+
+    return args.filename, args.vcf, args.out
+
+def check_snps(vcf, chromosome, left_bound, right_bound):
```

Comment:
```python
def check_snps(f_name, chromosome, left_bound, right_bound):
    vcf_in = VCF(f_name)
    # 1 is added to record.reference_start and the following parameter because vcf is 1 indexed
    # in order to keep code consistent
    region = '{c}:{l}-{r}'.format(c=chromosome, l=left_bound+1, r=right_bound+1)
    records = [rec for rec in vcf_in(region)]
    return records
```